### PR TITLE
release(dotnet-sdk): v0.2.2 w/ the OpenFgaClient & BatchCheck

### PR DESCRIPTION
**Pull request by rhamzeh**
## Description

- add OpenFgaClient wrapper
- implement `BatchCheck` to check multiple tuples in parallel
- implement `ListRelations` to check in one call whether a user has multiple relations to an objects
- add support for a non-transactional `Write`
- bump default max retries to `15`
- fix: retry on 5xx errors

## References

- Generates: https://github.com/openfga/dotnet-sdk/pull/14

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by adriantam — APPROVED** [review]

**Review by adriantam — DISMISSED** [review]

File: config/common/config.base.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
     ".github/workflows/semgrep.yaml": {}
   },
   "retryMaxAllowedNumber": 15,
-  "defaultMaxRetry": 5,
+  "defaultMaxRetry": 15,
```
Comment (adriantam) [inline config/common/config.base.json:90]: we bump this to 15 but in our change lot we say it is 5?

File: config/clients/dotnet/CHANGELOG.md.mustache [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
 
+## v0.2.2
+
+### [0.2.2](https://{{gitHost}}/{{gitUserId}}/{{gitRepoId}}/compare/v0.2.1...v0.2.2) (2023-04-12)
+
+- feat(client): add OpenFgaClient wrapper see [docs](https://{{gitHost}}/{{gitUserId}}/{{gitRepoId}}/tree/main#readme), see the `v0.2.1` docs for [the OpenFgaApi docs](https://{{gitHost}}/{{gitUserId}}/{{gitRepoId}}/tree/v0.2.1#readme)
+- feat(client): implement `BatchCheck` to check multiple tuples in parallel
+- feat(client): implement `ListRelations` to check in one call whether a user has multiple relations to an objects
+- feat(client): add support for a non-transactional `Write`
+- chore(config): bump default max retries to `5`
```
[the remark this answers is not in the document]
Comment (adriantam) [reply config/clients/dotnet/CHANGELOG.md.mustache:11]:
```suggestion
- chore(config): bump default max retries to `15`
```

**Review by adriantam — CHANGES_REQUESTED** [review]

File: config/clients/dotnet/template/api_test.mustache [hunk 3]
```diff
@@ -1237,6 +1420,46 @@ namespace {{testPackageName}}.Api {
             Assert.Equal(response, expectedResponse);
         }
 
+        /// <summary>
+        /// Test ListStores does not crash in error handling
```
[the remark this answers is not in the document]
Comment (rhamzeh) [reply config/clients/dotnet/template/api_test.mustache:1424]: We already do:
```csharp
var error = await Assert.ThrowsAsync<FgaApiNotFoundError>(ApiError);

            Assert.Equal(error.Method, HttpMethod.Get);
            Assert.Equal("ListStores", error.ApiName);
            Assert.Equal($"{_config.BasePath}/stores", error.RequestUrl);
            Assert.Null(error.StoreId);
```

[the remark this answers is not in the document]
Comment (adriantam) [reply config/clients/dotnet/template/api_test.mustache:1424]: In this case, maybe the description should be assert proper error is returned?

**Review by adriantam — DISMISSED** [review]

**Review by adriantam — APPROVED** [review]

**Pull request merged**